lib/_tls_common.js

```diff
@@ -153,6 +153,31 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const { privateKeyIdentifier, privateKeyEngine } = options;
+  if (privateKeyIdentifier !== undefined && privateKeyEngine !== undefined) {
+    if (key) {
+      // Both cleartext key and engine key can't be set at the same time
+      throw new ERR_INVALID_OPT_VALUE('privateKeyIdentifier',
+                                      privateKeyIdentifier);
+    }
+
+    if (typeof privateKeyIdentifier === 'string' &&
+        typeof privateKeyEngine === 'string') {
+      if (c.context.setEngineKey)
+        c.context.setEngineKey(privateKeyIdentifier, privateKeyEngine);
```

Comment:
I guess an empty string is allowed, `''`, because some HSMs might just map that to a default key?

Reply:
Yes, I think empty string for a key identifier should be fine (I haven't seen an engine that would do that in the wild though).